OtripleS.Web.Api.Tests.Unit/Services/SemesterCourseServiceTests/SemesterCourseServiceTests.Exceptions.Delete.cs

```diff
@@ -0,0 +1,165 @@
+// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using OtripleS.Web.Api.Models.SemesterCourses;
+using OtripleS.Web.Api.Models.SemesterCourses.Exceptions;
+using Xunit;
+
+namespace OtripleS.Web.Api.Tests.Unit.Services.SemesterCourseServiceTests
+{
+    public partial class SemesterCourseServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnDeleteWhenSqlExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid randomSemesterCourseId = Guid.NewGuid();
+            Guid inputSemesterCourseId = randomSemesterCourseId;
+            SqlException sqlException = GetSqlException();
+
+            var expectedSemesterCourseDependencyException =
+                new SemesterCourseDependencyException(sqlException);
+
+            this.storageBrokerMock.Setup(broker =>
+                    broker.SelectSemesterCourseByIdAsync(inputSemesterCourseId))
+                .ThrowsAsync(sqlException);
```

Comment:
Push this two tabs ahead

Reply:
I have tried to fix all tabs and spaces you asked. Please let me know If I did it wrong.